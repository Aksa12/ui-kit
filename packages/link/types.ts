export interface SharedLinkProps {
  /**
   * a url the user will be navigated to when clicking the button. This also changes the tag to `<a>`
   */
  url?: string;
  /**
   * if the `url` prop is set, this will open that link in a new tab
   */
  openInNewTab?: boolean;
}

export type LinkProps = SharedLinkProps & React.HTMLProps<HTMLAnchorElement>;
