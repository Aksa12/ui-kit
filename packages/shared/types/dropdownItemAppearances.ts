export enum DropdownItemAppearances {
  DANGER = "danger"
}
