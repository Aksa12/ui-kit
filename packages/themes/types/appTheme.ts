type Color = React.CSSProperties["color"];

export interface Theme {
  colors: {
    // Branding
    brandPrimary?: Color;
    brandPrimaryInverted?: Color;
    brandSecondary?: Color;
    brandSecondaryInverted?: Color;

    // States
    success?: Color;
    successInverted?: Color;
    warning?: Color;
    warningInverted?: Color;
    error?: Color;
    errorInverted?: Color;

    // Backgrounds
    bgPrimary?: Color;
    bgPrimaryInverted?: Color;
    bgSecondary?: Color;
    bgSecondaryInverted?: Color;
    bgHover?: Color;
    bgHoverInverted?: Color;
    bgScrim?: Color;
    bgScrimInverted?: Color;
    bgSelected?: Color;
    bgSelectedInverted?: Color;
    bgAppHeader?: Color;

    // Text
    textColorPrimary?: Color;
    textColorPrimaryInverted?: Color;
    textColorSecondary?: Color;
    textColorSecondaryInverted?: Color;
    textColorInteractive?: Color;
    textColorInteractiveInverted?: Color;

    // Decorators
    border?: Color;
    borderInverted?: Color;
    borderHeavy?: Color;
    borderHeavyInverted?: Color;
    shadow?: Color;
  };
}
