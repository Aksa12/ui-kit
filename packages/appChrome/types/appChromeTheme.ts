import { Theme } from "../../themes/types/appTheme";

type BgColor = React.CSSProperties["backgroundColor"];
type PaddingHoriz =
  | React.CSSProperties["paddingLeft"]
  | React.CSSProperties["paddingRight"];
type PaddingVert =
  | React.CSSProperties["paddingBottom"]
  | React.CSSProperties["paddingTop"];
type ElWidth = React.CSSProperties["width"];

export interface AppChromeTheme extends Theme {
  headerBackgroundColor?: BgColor;
  headerPaddingHor?: PaddingHoriz;
  headerPaddingVert?: PaddingVert;
  iconWidth?: ElWidth;
  itemHoverBackgroundColor?: BgColor;
  sidebarBackgroundColor?: BgColor;
  sidebarHeaderPaddingHor?: PaddingHoriz;
  sidebarHeaderPaddingVert?: PaddingVert;
  sidebarWidth?: ElWidth;
  sidebarItemPaddingHor?: PaddingHoriz;
  sidebarItemPaddingVert?: PaddingVert;
}
