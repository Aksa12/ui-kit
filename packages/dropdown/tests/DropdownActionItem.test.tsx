import React from "react";
import { mount } from "enzyme";
import * as emotion from "emotion";
import { createSerializer } from "jest-emotion";
import toJson from "enzyme-to-json";
import { DropdownItemAppearances } from "../../shared/types/dropdownItemAppearances";

import { DropdownActionItem } from "../";

expect.addSnapshotSerializer(createSerializer(emotion));

describe("Dropdown", () => {
  it("default", () => {
    const component = mount(
      <DropdownActionItem key="edit" value="edit">
        Edit
      </DropdownActionItem>
    );

    expect(toJson(component)).toMatchSnapshot();
  });
  it("default", () => {
    Object.keys(DropdownItemAppearances).forEach(appearance => {
      const component = mount(
        <DropdownActionItem
          key="edit"
          value="edit"
          appearance={DropdownItemAppearances[appearance]}
        >
          Edit
        </DropdownActionItem>
      );

      expect(toJson(component)).toMatchSnapshot();
    });
  });
});
