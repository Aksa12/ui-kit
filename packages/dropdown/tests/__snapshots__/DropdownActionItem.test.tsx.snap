// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown default 1`] = `
<DropdownActionItem
  value="edit"
>
  Edit
</DropdownActionItem>
`;

exports[`Dropdown default 2`] = `
<DropdownActionItem
  appearance="danger"
  value="edit"
>
  Edit
</DropdownActionItem>
`;
